PatientWebApplication/SearchMicroserviceApi/Service/AppointmentSearchService.cs

```diff
@@ -0,0 +1,216 @@
+            }
+            else if (!UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                appointments = GetAppointmentsAfterDate(appointmentSearchDto.Start, appointments);
+            }
+            return appointments;
+
+        }
+
+        /// <summary> This method is getting list of filtered <c>DoctorAppointment</c> of one patient that are between two dates. </summary>
+        /// <param name="appointments"><c>appointments</c> is List of appointments that matches search fields.
+        /// </param>
+        /// <param name="start"><c>start</c> is first date of search.
+        /// </param>
+        /// <param name="end"><c>end</c> is last date of search.
+        /// </param>
+        /// <returns> List of filtered patient's appointments. </returns>
+        private List<AppointmentDto> GetAppointmentsBetweenDates(string start, string end, List<AppointmentDto> appointments)
+        {
+            DateTime startDate = UtilityMethods.ParseDateInCorrectFormat(start);
+            DateTime endDate = UtilityMethods.ParseDateInCorrectFormat(end);
+            return appointments.FindAll(appointment => startDate <= UtilityMethods.ParseDateInCorrectFormat(appointment.Date) && UtilityMethods.ParseDateInCorrectFormat(appointment.Date) <= endDate);
+        }
+
+        /// <summary> This method is getting list of filtered <c>DoctorAppointment</c> of one patient that are before given date. </summary>
+        /// <param name="appointments"><c>appointments</c> is List of appointments that matches search fields.
+        /// </param>
+        /// <param name="date"><c>date</c> is last date of search.
+        /// </param>
+        /// <returns> List of filtered patient's appointments. </returns>
+        private List<AppointmentDto> GetAppointmentsBeforeDate(string date, List<AppointmentDto> appointments)
+        {
+            DateTime endDate = UtilityMethods.ParseDateInCorrectFormat(date);
+            return appointments.FindAll(appointment => UtilityMethods.ParseDateInCorrectFormat(appointment.Date) <= endDate);
+        }
+
+        /// <summary> This method is getting list of filtered <c>DoctorAppointment</c> of one patient that are after given date. </summary>
+        /// <param name="appointments"><c>appointments</c> is List of appointments that matches search fields.
+        /// </param>
+        /// <param name="date"><c>date</c> is first date of search.
+        /// </param>
+        /// <returns> List of filtered patient's appointments. </returns>
+        private List<AppointmentDto> GetAppointmentsAfterDate(string date, List<AppointmentDto> appointments)
+        {
+            DateTime startDate = UtilityMethods.ParseDateInCorrectFormat(date);
+            return appointments.FindAll(appointment => startDate <= UtilityMethods.ParseDateInCorrectFormat(appointment.Date));
```

Comment:
Write this in form:
"
return appointments.FindAll(appointment => UtilityMethods.ParseDateInCorrectFormat(date) <= UtilityMethods.ParseDateInCorrectFormat(appointment.Date));
"